teamcity: failed tests on master: test (proposer evaluated kv)/TestRemovedReplicaError

The following tests appear to have failed:

[#108341](https://teamcity.cockroachdb.com/viewLog.html?buildId=108341):

```
--- FAIL: test (proposer evaluated kv)/TestRemovedReplicaError (0.002s)
	client_raft_test.go:2723: expected RangeNotFoundError; got range 1: replica {0 0 0} not lease holder; lease holder unknown

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/storage/replica.go
+++ b/pkg/storage/replica.go
@@ -1024,13 +1024,16 @@ func (r *Replica) redirectOnOrAcquireLease(ctx context.Context) (LeaseStatus, *r
 				select {
 				case pErr = <-llChan:
 					if pErr != nil {
-						// Getting a LeaseRejectedError back means someone else got there
-						// first, or the lease request was somehow invalid due to a
-						// concurrent change. Convert the error to a NotLeaseHolderError.
+						// Getting a LeaseRejectedError back either means that someone else
+						// got there first or that the lease request was somehow invalid due
+						// to a concurrent change. That concurrent change could have been
+						// that this replica was removed, so check for that case before
+						// falling back to a NotLeaseHolderError.
 						if _, ok := pErr.GetDetail().(*roachpb.LeaseRejectedError); ok {
-							lease, _ := r.getLease()
 							var err error
-							if !r.IsLeaseValid(lease, r.store.Clock().Now()) {
+							if _, descErr := r.GetReplicaDescriptor(); descErr != nil {
+								err = descErr
+							} else if lease, _ := r.getLease(); !r.IsLeaseValid(lease, r.store.Clock().Now()) {
 								err = newNotLeaseHolderError(nil, r.store.StoreID(), r.Desc())
 							} else {
 								err = newNotLeaseHolderError(lease, r.store.StoreID(), r.Desc())

